lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -284,12 +293,23 @@ class BuildOpGroups : public calyx::FuncOpPartialLoweringPattern {
     IRRewriter::InsertionGuard guard(rewriter);
     rewriter.setInsertionPointToEnd(group.getBody());
     auto addrPorts = memoryInterface.addrPorts();
-    assert(addrPorts.size() == addressValues.size() &&
-           "Mismatch between number of address ports of the provided memory "
-           "and address assignment values");
-    for (auto address : enumerate(addressValues))
-      rewriter.create<calyx::AssignOp>(loc, addrPorts[address.index()],
-                                       address.value());
+    if (addressValues.empty()) {
+      assert(
+          addrPorts.size() == 1 &&
+          "We expected a 1 dimensional memory of size 1 because there were no "
+          "address assignment values");
+      // Assign 1'd0 to the address port.
```

Comment:
Is it legal to use a `0'd0` in Calyx? Other pipelines (e.g. Chisel -> FIRRTL -> CIRCT) deal with a similar edge case, where a 1 element vector has a 0 width wire to index into it. I guess it just matters that this is aligned with what Calyx expects, and if it expects a 1 bit address, then I guess that's fine.

Reply:
Yeah, Calyx doesn't allow a 0 width wire, so we have to make it 1 bit.